src/index.ts

```diff
@@ -84,22 +87,67 @@ const SchemaExtensionPlugin = makeExtendSchemaPlugin(build => {
           const {
             graphile: { fieldContext },
           } = resolveInfo;
-          return representations.map((representation: any) => {
+          return representations.map(async (representation: any) => {
             if (!representation || typeof representation !== "object") {
               throw new Error("Invalid representation");
             }
+
             const { __typename, [nodeIdFieldName]: nodeId } = representation;
-            if (!__typename || typeof nodeId !== "string") {
-              throw new Error("Failed to interpret representation");
+            if (!__typename) {
+              throw new Error(
+                "Failed to interpret representation, no typename"
+              );
+            }
+            if (nodeId) {
+              if (typeof nodeId !== "string") {
+                throw new Error(
+                  "Failed to interpret representation, invalid nodeId"
+                );
+              }
+              const x = resolveNode(
+                nodeId,
+                build,
+                fieldContext,
+                data,
+                context,
+                resolveInfo
+              );
+
+              return x;
+            } else {
+              const type = getTypeByName(__typename);
+              const { pgIntrospection: table } = scopeByType.get(type);
+
+              if (!table.primaryKeyConstraint) {
+                throw new Error("Failed to interpret representation");
+              }
+              const {
+                primaryKeyConstraint: { keyAttributes },
+              } = table;
+
+              const whereClause = sql.fragment`(${sql.join(
+                keyAttributes.map(
+                  (attr: PgAttribute) =>
+                    sql.fragment`${sql.identifier(attr.name)} = ${sql.value(
+                      representation[inflection.column(attr)]
+                    )}`
+                ),
+                ") and ("
+              )})`;
+
+              const rows = await resolveInfo.graphile.selectGraphQLResultFromTable(
+                sql.identifier(table.namespace, table.name),
+                (_alias, queryBuilder) => {
+                  queryBuilder.where(whereClause);
+                }
+              );
+
+              if (rows.count !== 1) {
```

Comment:
I think this should be `rows.length`?
```suggestion
              if (rows.length !== 1) {
```